core/arch/interp.c

```diff
@@ -2852,6 +2852,13 @@ client_process_bb(dcontext_t *dcontext, build_bb_t *bb)
         if (!instr_opcode_valid(inst))
             continue;
 
+# ifdef AARCH64
+        if (inst != instrlist_last(bb->ilist)) {
+            CLIENT_ASSERT(instr_get_opcode(inst) != OP_isb,
```

Comment:
nit: the order seems odd: I would vote for what IMHO is easier to read and matches the existing code: more like "if (instr_get_opcode(inst) == OP_isb) CLIENT_ASSERT(inst == instrlist_last..."